src/flag_gems/ops/layernorm.py

```diff
@@ -237,6 +237,8 @@ def layer_norm_backward_kernel(
     BLOCK_COL_SIZE: tl.constexpr,
 ):
     pid = tle.program_id(0) * BLOCK_ROW_SIZE + tl.arange(0, BLOCK_ROW_SIZE)[:, None]
+    if pid >= M:
```

Comment:
`pid` here is a tensor but not a scalar

Reply:
We can only return when the condition is a scalar comparison

Reply:
maybe our naming mislead you. we should rename it as offset instead of pid here.